syft/pkg/cataloger/dotnet/parse_dotnet_packages_lock.go

```diff
@@ -0,0 +1,130 @@
+package dotnet
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"sort"
+
+	"github.com/anchore/packageurl-go"
+	"github.com/anchore/syft/internal/log"
+	"github.com/anchore/syft/internal/relationship"
+	"github.com/anchore/syft/syft/artifact"
+	"github.com/anchore/syft/syft/file"
+	"github.com/anchore/syft/syft/pkg"
+	"github.com/anchore/syft/syft/pkg/cataloger/generic"
+)
+
+var _ generic.Parser = parseDotnetPackagesLock
+
+type dotnetPackagesLock struct {
+	Version      int                                         `json:"version"`
+	Dependencies map[string]map[string]dotnetPackagesLockDep `json:"dependencies"`
```

Comment:
Comment for other reviewers - the reason this looks this way is that the dependencies have this nested map quality where there are sub fields. @Kemosabert do you have an example where we can see how a sibling to the top level package would behave? In the bottom example it would be a sibling package to `netcore2.0` - I pulled this from [here](https://github.com/NuGet/Home/wiki/Enable-repeatable-package-restore-using-lock-file)


```
  "dependencies": {	
    "netcore2.0": {	
      "Contoso.Base": {	
        "type": "direct",	
        "requested": "3.0.0",	
        "resolved": "3.0.0",
        "contentHash":"fVXsnMP2Wq84VA533zj0a/Et+QoLoeNpVXsnMP2Wq84l+hsUxfwunkbqoIHIvpOqwQ/+HIvprVKs+QOihnkbqod=="
        "dependencies": {
             "Contoso.Core": "1.2.3",
             "Fabrikam.Utilities": "[3.1.0]"
         }		
      }	
      "Contoso.Core": {
        "type": "transitive",	
        "requested": "1.2.3",	
        "resolved": "1.2.3",
        "contentHash":"xScnMP2Wq84VA533zj0a/Et+QoLoeNpVXsnMP2Wq84l+hsUxfwunkbqoIHIvpOqwQ/+HIvprVKs+QOihnkbmoq=="
        "dependencies": {
           ...
           ...
         }
```

Reply:
As mentioned below, `.netcore2.0` is the runtime target version for the deps. I am not quite sure how to let Syft describe this in the SBOM since it's not really a package, its just to tell nuget which packages to install for the user's runtime of .NET.

My first thought would be to let the SBOM include all packages from all runtime versions, and show the runtime versions where they appear in either in the locations or in the metadata. What do you think?